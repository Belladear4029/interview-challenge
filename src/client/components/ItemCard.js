// @flow
import React from "react";
import { render } from "react-dom";

type Props = {
  item: {
    id: number,
    name: string,
    dietaries: Array<string>,
  },
  removeItem?: Function,
};

const ItemCard = ({ item, removeItem }: Props): React$Element<"div"> => {
  const { id, name, dietaries } = item;

  return (
    <div className="item-card">
      <h2>{name}</h2>
      <p>
        {dietaries.map((dietary, i) => (
          <span key={i} className="dietary">
            {dietary}
          </span>
        ))}
      </p>
      {removeItem && (
        <button onClick={() => removeItem(id)} className="remove-item">
          x
        </button>
      )}
    </div>
  );
};

export default ItemCard;
