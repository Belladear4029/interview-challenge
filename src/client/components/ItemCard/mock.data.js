export const item = {
  id: 1001,
  name: "Kale Caesar Pasta, Turmeric Satay Broccoli & Lemon Cashew Greens",
  dietaries: ["v", "ve", "df", "gf", "n!"],
};
