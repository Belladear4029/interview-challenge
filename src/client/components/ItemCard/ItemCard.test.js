import * as React from "react";
import { shallow, mount } from "enzyme";
import ItemCard from "./ItemCard";
import { item } from "./mock.data";

const removeItem = () => {};

const shallowDefault = (props) =>
  shallow(<ItemCard {...props} removeItem={(event) => removeItem(event)} />);

const mountDefault = (props) =>
  mount(<ItemCard {...props} removeItem={(event) => removeItem(event)} />);

describe("<ItemCard>", () => {
  it("renders correctly", () => {
    const wrapper = shallowDefault({ item });
    expect(wrapper).toMatchSnapshot();
  });
  it("renders correct props", () => {
    const wrapper = mountDefault({ item });
    expect(wrapper.props().item).toEqual(item);
  });
});
