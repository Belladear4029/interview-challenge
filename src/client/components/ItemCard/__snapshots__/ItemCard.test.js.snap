// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemCard> renders correctly 1`] = `ShallowWrapper {}`;
